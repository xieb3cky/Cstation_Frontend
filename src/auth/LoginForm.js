import React, { useState } from "react";
import { useNavigate } from "react-router-dom";
// import Alert from "../common/Alert";


/** Login form.
 *
 * Shows form and manages update to state on changes.
 * On submission:
 * - calls login function prop
 * - redirects to /companies route
 *
 * Routes -> LoginForm -> Alert
 * Routed as /login
 */

function LoginForm({ login }) {
    const navigate = useNavigate();
    const [formData, setFormData] = useState({
        username: "",
        password: "",
    });
    const [formErrors, setFormErrors] = useState([]);

    console.debug(
        "LoginForm",
        "login=", typeof login,
        "formData=", formData,
        "formErrors", formErrors,
    );

    /** Handle form submit:
     *
     * Calls login func prop and, if successful, redirect to /companies.
     */

    async function handleSubmit(evt) {
        evt.preventDefault();
        let result = await login(formData);

        if (result.success) {
            navigate("/");
        } else {
            setFormErrors(result.errors);
        }
        setFormData({
            username: "",
            password: ""
        })
    }

    /** Update form data field */
    function handleChange(evt) {
        const { name, value } = evt.target;
        setFormData(l => ({ ...l, [name]: value }));
    }

    return (
        <div className="flex center body-container">
            <div className="page-grid grid">
                <section className="banner grid center">
                </section>

                <section className="form-wrapper grid center">
                    <form className="form-section" onSubmit={handleSubmit}>
                        <div className="header">
                            <h1>Welcome Back ! ðŸ˜„</h1>
                            <p>Please enter your details.</p>
                        </div>
                        <div className="field">
                            <label>Username</label>
                            <input
                                name="username"
                                className="form-control"
                                value={formData.username}
                                onChange={handleChange}

                            />
                        </div>
                        <div className="field">
                            <label>Password</label>
                            <input
                                type="password"
                                name="password"
                                className="form-control"
                                value={formData.password}
                                onChange={handleChange}

                            />
                        </div>

                        <button type="submit" className="btn btn-primary" onSubmit={handleSubmit}>Sign in</button>
                    </form>
                </section>


            </div>
        </div>
    );
}

export default LoginForm;

