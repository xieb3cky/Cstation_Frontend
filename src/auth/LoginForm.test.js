import React from "react";
import { render, fireEvent, waitFor } from "@testing-library/react";
import LoginForm from "./LoginForm";
import { MemoryRouter } from "react-router";

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <LoginForm />
    </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});

it('calls login function prop and redirects on successful submission', async () => {
  const loginMock = jest.fn(() => Promise.resolve({ success: true }));
  const { getByLabelText, getByText } = render(
    <MemoryRouter>
      <LoginForm login={loginMock} />
    </MemoryRouter>
  );

  const usernameInput = getByLabelText('Username');
  const passwordInput = getByLabelText('Password');
  const submitButton = getByText('Sign in');

  fireEvent.change(usernameInput, { target: { value: 'testuser' } });
  fireEvent.change(passwordInput, { target: { value: 'testpassword' } });
  fireEvent.click(submitButton);

  await waitFor(() => {
    expect(loginMock).toHaveBeenCalledWith({ username: 'testuser', password: 'testpassword' });
    expect(window.location.pathname).toBe('/');
  });
});