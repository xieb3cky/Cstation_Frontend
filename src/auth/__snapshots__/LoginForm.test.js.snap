// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex center body-container"
  >
    <div
      class="page-grid grid"
    >
      <section
        class="banner grid center"
      >
        <section
          class="banner grid center"
        >
          <div
            class="road"
          >
            <div
              class="taxi"
            >
              <div
                class="light_beam"
              />
              <div
                class="side_mirror"
              />
              <span>
                <b />
                <i />
              </span>
            </div>
          </div>
        </section>
      </section>
      <section
        class="form-wrapper grid center"
      >
        <form
          class="form-section"
        >
          <div
            class="header"
          >
            <h1>
              Welcome Back ! ðŸ˜„
            </h1>
            <p>
              Please enter your details.
            </p>
          </div>
          <div
            class="field"
          >
            <label>
              Username
            </label>
            <input
              class="form-control"
              name="username"
              value=""
            />
          </div>
          <div
            class="field"
          >
            <label>
              Password
            </label>
            <input
              class="form-control"
              name="password"
              type="password"
              value=""
            />
          </div>
          <button
            class="btn btn-primary"
            type="submit"
          >
            Sign in
          </button>
        </form>
      </section>
    </div>
  </div>
</DocumentFragment>
`;
